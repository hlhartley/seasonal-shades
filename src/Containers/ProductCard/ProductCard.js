import React, { Component } from 'react';
import { connect } from 'react-redux';

export class ProductCard extends Component {
    render() {
        return(
            <div>
                <div className='product-card-container'>
                    <h2>MATCHING PRODUCT:</h2> 
                    <h2>{this.props.allColors[this.props.color].product.name}</h2>
                    <p>{this.props.allColors[this.props.color].product.description}</p>
                </div>
                <div className='product-info'>
                    <div>
                        <img src={`${this.props.allColors[this.props.color].product.image_link}`} className='product-image'/>
                    </div>
                    <div className='product-info-text'>
                        <p><b>Brand:</b> {this.props.allColors[this.props.color].product.brand}</p>
                        <p><b>Price:</b> ${this.props.allColors[this.props.color].product.price}</p>
                        <p><b>Website:</b> {this.props.allColors[this.props.color].product.website_link}</p>
                    </div>
                </div>
            </div>
        )
    }
}

export const mapStateToProps = (state) => ({
    nail_polishColors: state.nailpolishColors,
    lipstickColors: state.lipstickColors,
    blushColors: state.blushColors,
    eyeshadowColors: state.eyeshadowColors,
    allColors: state.allColors,
});


export default connect(mapStateToProps)(ProductCard);
