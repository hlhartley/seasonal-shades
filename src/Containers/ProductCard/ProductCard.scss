.product-card-container {
    display: flex;
    flex-direction: column;
}

.product-image {
    height: 150px;
    margin-top: 20px;
    width: 150px;
}

.product-info {
    align-items: center;
    display: flex;
    justify-content: center;
}

.product-info-text {
    text-align: left;
}