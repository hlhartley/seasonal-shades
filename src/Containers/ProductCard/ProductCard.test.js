import React from 'react';
import { shallow } from 'enzyme';
import ProductCard, { mapStateToProps } from './ProductCard';

describe('ProductCard', () => {
    let wrapper;
    beforeEach(() => {
        const allColors = {
            522: {
                product: {},
                hexcode: '#EED8BE'
            },
            771: {
                product: {},
                hexcode: '#BD9E9B' 
            },
            'casablanca': {
                product: {},
                hexcode: '#444446'
            }
        }
        wrapper = shallow(<ProductCard color={'Casablanca'} allColors={allColors} lipstickColors={[]} blushColors={[]} eyeshadowColors={[]} nail_polishColors={[]}/>)
    });
    describe('ProductCard component', () => {
        it('should have the correct snapshot with the correct data passed in', () => {
            expect(wrapper).toMatchSnapshot()
        });
    });

    describe('mapStateToProps', () => {
        it('should return an object with all colors', () => {
            const mockState = {
                allColors: {
                    522: {
                        product: {},
                        hexcode: '#EED8BE'
                    },
                    771: {
                        product: {},
                        hexcode: '#BD9E9B' 
                    },
                    'casablanca': {
                        product: {},
                        hexcode: '#444446'
                    }
                },
                blushColors: undefined,
                eyeshadowColors: undefined,
                lipstickColors: [],
                nail_polishColors: undefined,
                moreColors: undefined,
            }
            const expected = {
                allColors: {
                    522: {
                        product: {},
                        hexcode: '#EED8BE'
                    },
                    771: {
                        product: {},
                        hexcode: '#BD9E9B' 
                    },
                    'casablanca': {
                        product: {},
                        hexcode: '#444446'
                    }
                },
                blushColors: undefined,
                eyeshadowColors: undefined,
                lipstickColors: [],
                nail_polishColors: undefined,
            }
            const mappedProps = mapStateToProps(mockState)
            expect(mappedProps).toEqual(expected)
        });
    });
});