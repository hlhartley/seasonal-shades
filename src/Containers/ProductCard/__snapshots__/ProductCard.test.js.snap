// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard ProductCard component should have the correct snapshot with the correct data passed in 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
